openedx/openedx-platform
# feat: emit passing status updated events for badging

## Description

This PR introduces emission of the `COURSE_PASSING_STATUS_UPDATED` as well as `CCX_COURSE_PASSING_STATUS_UPDATED` events, that are groundwork for the new Credly integration and the future badging initiative.


Product GH ticket for tracking - https://github.com/openedx/platform-roadmap/issues/280

## Supporting information

Link to other information about the change, such as Jira issues, GitHub issues, or Discourse discussions.
Be sure to check they are publicly readable, or if not, repeat the information here.

## Testing instructions

- Configure Redis stream or Kafka event bus
- Set `settings.FEATURES['BADGES_ENABLED']` to True
- Configure grading in a course
- Pass the course with testing user
- Observe generated COURSE_PASSING_STATUS_UPDATED events
- Change grading for this course in a studio
- Observe generated COURSE_PASSING_STATUS_UPDATED events

(The same should work for CCX courses)

Please refer to openedx-events PR for more information on the events itself: https://github.com/openedx/openedx-events/pull/303

## Deadline

Redwood

## Other information

This PR is the replacement for the https://github.com/openedx/edx-platform/pull/34524
